src/CabanaPD_Force.hpp

```diff
@@ -668,42 +667,43 @@ class Force<ExecutionSpace, ForceModel<LPS, Fracture>>
                                                                   i );
             for ( std::size_t n = 0; n < num_neighbors; n++ )
             {
-                if ( mu( i, n ) > 0 )
+                double fx_i = 0.0;
+                double fy_i = 0.0;
+                double fz_i = 0.0;
+
+                std::size_t j =
+                    Cabana::NeighborList<NeighListType>::getNeighbor(
+                        neigh_list, i, n );
+
+                // Get the reference positions and displacements.
+                double xi, r, s;
+                double rx, ry, rz;
+                getDistanceComponents( x, u, i, j, xi, r, s, rx, ry, rz );
+
+                // Break if beyond critical stretch unless in no-fail zone.
+                if ( r * r >= break_coeff * xi * xi && !nofail( i ) &&
+                     !nofail( i ) )
+                {
+                    mu( i, n ) = 0;
+                    // Note m( i ) = 0 when all bonds connected to particle i
```

Comment:
I would rather this comment be in the `else if` block
"Check if this bond is broken (mu=0) to ensure both m(i) and m(j) are both >0 (which only occurs when all bonds are broken) to avoid divide by zero."